internal/rbac/main.go

```diff
@@ -90,6 +71,30 @@ func StudyIDsWithRole(user types.User, studyRoleName StudyRoleName) ([]uuid.UUID
 	return studyIDs, nil
 }
 
+// Add a project TRE owner role for a user
+func AddProjectTreOwnerRole(user types.User, projectId uuid.UUID) (bool, error) {
+	roleName := makeProjectOwnerRole(projectId).RoleName()
+	return addOwnerRoleForResource(user, roleName, []string{"/projects/tre"}, projectId)
+}
+
+// Project IDs where a user has a role
+func ProjectIDsWithRole(user types.User, projectRoleName ProjectRoleName) ([]uuid.UUID, error) {
```

Comment:
This can potentially be abstracted to combine the logic with `StudyIDsWithRole`. But I wanted to hear peoples thoughts first as I'm not sure if the project environments complicates this or not, and therefore whether separating the logic is overall more simple to follow?